src/SLCore/Core/IRpcDebugger.cs

```diff
@@ -43,11 +43,17 @@ public RpcDebugger() : this(new FileSystem(), DateTime.Now)
     {
     }
 
-    internal /* for testing */ RpcDebugger(IFileSystem fileSystem, DateTime dateTime)
+    internal /* for testing */ RpcDebugger(string fileNameOverride) : this(new FileSystem(), DateTime.Now, fileNameOverride)
```

Comment:
You did it kind of backwards

```
RpcDebugger() : this(new FileSystem(), DateTime.Now)
RpcDebugger(IFileSystem fileSystem, DateTime fileDate) : this(fileSystem, $"{dateTime.ToString("yyyy-MM-dd_HHmmssffff")}.log")
RpcDebugger(IFileSystem fileSystem, string fileName)
```

but at the same time, the file will end up in the common folder for integration tests, which is not what we want, so:


```
RpcDebugger() : this(new FileSystem(), DateTime.Now)
RpcDebugger(IFileSystem fileSystem, DateTime fileDate) : this(fileSystem, $"{dateTime.ToString("yyyy-MM-dd_HHmmssffff")}.log", Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "SonarLint for Visual Studio", "Rpc Logs"))
RpcDebugger(IFileSystem fileSystem, string fileName, string baseFolder)
```

The integration tests then would initialize it as `new(new FileSystem(), "logrpc.txt", "privateFolder")`

Reply:
I think integration test should mimic the behaviour as much as possible and having them on the common folder is not an issue. Only when we run the tests we would create these files and it won't create tons of file that would make it harder when we really run the logs.